samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -97,7 +126,13 @@ void handleContainerLaunch(SamzaResourceRequest request, String preferredHost, S
    */
   void handleContainerStop(String processorId, String containerId, String preferredHost, int exitStatus,
       Duration preferredHostRetryDelay, ContainerAllocator containerAllocator) {
-    if (standbyContainerManager.isPresent()) {
+    if (getControlActionMetadata(processorId).isPresent()) {
+      LOG.info("Setting the container with processorId {} stopped to be true because of existing container placement action", processorId);
+      Object activeContainerStopped = getControlActionMetadata(processorId).get().getActiveContainerStopped();
+      synchronized (activeContainerStopped) {
+        activeContainerStopped.notifyAll();
```

Comment:
What are we trying to accomplish by having this notify here?

Reply:
ContainerAllocator thread used to wait and get notified by the ClusterResourceManager callback thread on stop notification receiving, but we are gonna change to do busy looping with a sleep timeout so when later we parallelize container allocation, the allocation threads would not need to be blocked by this change.